### PR TITLE
chore: remove google/go-github, google/uuid, gookit/color dependencies

### PR DESCRIPTION
**Pull request by eunyoung14**
## Summary

Replace three heavyweight dependencies with minimal inline implementations to reduce the dependency tree and attack surface.

## Changes

| Dependency | Usage | Replacement |
|-----------|-------|-------------|
| `google/go-github/v71` | Version check (1 API call) | Plain `net/http` GET to GitHub API |
| `google/uuid` | `IsUUID()` (1 function) | Regex: all-dashes or 32-char hex |
| `gookit/color` | 5 color functions | ANSI escape codes (~30 lines) |

### Details

- **`cmd/version.go`**: Replace `github.NewClient` with a direct `net/http` request to `api.github.com/repos/alpacax/alpacon-cli/releases/latest`. Adds 5s timeout and `io.LimitReader` (1 MiB) for defense-in-depth
- **`utils/utils.go`**: Replace `uuid.Parse()` with strict regex matching standard UUID format (with dashes) or 32-char hex (without dashes). No mixed-dash inputs accepted
- **`utils/color.go`**: Replace `gookit/color` with raw ANSI escape sequences. Same visual behavior, zero dependencies

### Impact

Removes from `go.mod`:
- `google/go-github/v71` + `google/go-querystring`
- `google/uuid`
- `gookit/color` + `xo/terminfo`

## Checklist

- [x] Code follows project style guidelines
- [x] Self-reviewed the code
- [x] No hardcoded secrets or credentials
- [x] Error handling is complete
- [x] Tests include edge cases (IsUUID tests pass with new regex)
- [x] `go mod tidy` executed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/version.go [hunk 1]
```diff
@@ -15,23 +18,48 @@ var versionCmd = &cobra.Command{
...
+	req.Header.Set("User-Agent", utils.GetUserAgent())
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", "", err
+	}
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", "", fmt.Errorf("GitHub API returned %d", resp.StatusCode)
```
[the remark this answers is not in the document]
Comment (eunyoung14) [reply cmd/version.go]: Fixed. Now uses resp.Status (e.g., "403 Forbidden") instead of just the numeric code.
